app/models/eds/repository.rb

```diff
@@ -41,5 +41,21 @@ def eds_find(id, params, eds_params)
                                            document_model: blacklight_config.document_model,
                                            blacklight_config: blacklight_config)
     end
+
+    # Some data providers in EDS don't have the data that populates the pub_year_tisim field
+    # (although we can query against the data), so this forces the pub_year_tisim field to be populated
+    # when the response from EDS still contains date range data.
+    # https://github.com/sul-dlss/SearchWorks/issues/1721
+    def eds_results_with_date_range(results)
+      return results if results.dig('facet_counts', 'facet_fields', 'pub_year_tisim').present?
+      return results unless results.key?('date_range')
```

Comment:
We could just return the `date_range` `minyear` and `maxyear` unconditionally when available, but this felt potentially a bit safer.

Reply:
Yeah I get that.